test/correctness/math.cpp

```diff
@@ -29,6 +37,15 @@ bool relatively_equal(value_t a, value_t b) {
             return true;
         }
 
+        // try again with a lower error threshold in case the device target is
+        // using fast approximate trigonometric functions (HLSL):
+        if (relative_error < .001023) {
```

Comment:
Prolly should document why this threshold was chosen. I'm not quite sure I understand what is going on here. There's not much use in extra output from a test that doesn't fail and I think this will allow the test to pass on all targets with the lower threshold, which is probably not desired.

Reply:
I found that number empirically. I could not find any documentation describing the numerical precision and limitations of the trigonometric functions and algorithms endorsed by HLSL.

Should we extend `relatively_equal()` by adding an extra parameter for the tolerance (which is then decided based on the target)?

Reply:
I added `target` as a parameter to `relatively_equal()` and then check for `target.supports_device_api(DeviceAPI::D3D12Compute)` before allowing this relaxed relative error threshold. Is this acceptable?